src/engine-actions/wait-for-swap-commitment.js

```diff
@@ -12,25 +12,50 @@ class CanceledSwapError extends Error {}
  */
 async function waitForSwapCommitment (swapHash) {
   return new Promise((resolve, reject) => {
+    // we use an array for timers since an invoice can be
+    // trigger the stream in an OPEN state multiple times.
+    const timers = []
+
     // subscribeSingleInvoice always sends out the initial invoice state
     const stream = subscribeSingleInvoice(swapHash, { client: this.client })
-    stream.on('end', () => reject(new Error(
-      'Stream ended while waiting for commitment on ' + swapHash)))
+    stream.on('end', () => {
+      stream.removeAllListeners()
+      reject(new Error('Stream ended while waiting for commitment on ' + swapHash))
+
+      // stop the timers on invoice expiration if they are still active
+      timers.forEach(clearTimeout)
+    })
     stream.on('data', function (invoice) {
       switch (invoice.state) {
         case INVOICE_STATES.OPEN:
+          const creationDate = new Date(invoice.creationDate * 1000)
+          const expirationDate = new Date(
+            creationDate.getTime() + (invoice.expiry * 1000)
+          )
+
+          // if the invoice is open but still expired, we consider it canceled
```

Comment:
So LND still calls an invoice "open" after expiration?

Reply:
Should "still expired" be "already expired"?

Reply:
Yes, the state of the invoice is managed separately from whether it is expired. Cancellation is one reason an invoice will be rejected, expiration is another.